pkg/siteconfig/clusterinstance.go

```diff
@@ -162,9 +268,233 @@ func (builder *ClusterInstanceBuilder) Exists() bool {
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
+// WaitForCIVCondition waits until the ClusterInstance type:ClusterInstanceValidated
+// has a condition that matches the expected, checking only the Type, Status, Reason, and Message fields.
+// For the message field, it matches if the message contains the expected.
+// Zero fields in the expected condition are ignored.
+func (builder *CIBuilder) WaitForCIVCondition(expected metav1.Condition, timeout time.Duration) (*CIBuilder, error) {
```

Comment:
@mpmaruthu It looks like the actual type of the condition you're waiting for changes in the name but not the code for the function. Could you just have one `WaitForCondition` function and then use conditions as variables?

Reply:
Updated. Thanks!